src/components/course/edit/sections/LecturerSection.vue

```diff
@@ -1,125 +1,121 @@
+                        >
+                            @{{ currentSelection.value.username }}
+                        </router-link>
+                        )
+                    </label>
+                    <label v-else class="input-label-error">
+                        <i class="text-red-400 fas fa-times mr-2"></i>
+                        Lecturer-ID not found!
+                    </label>
                 </div>
             </div>
+            <div class="lg:w-1/2 w-full invisible" />
         </div>
-    </section>
+    </BaseSection>
 </template>
 
 <script lang="ts">
-    import { computed, onBeforeMount, ref, watch } from "vue";
+    import SearchableSelect from "@/components/common/SearchableSelect.vue";
+    import BaseSection from "@/components/common/section/BaseSection.vue";
+    import { computed, onBeforeMount, ref } from "vue";
+    import SearchSelectOption from "@/use/helpers/SearchSelectOption";
     import Lecturer from "@/api/api_models/user_management/Lecturer";
     import UserManagement from "@/api/UserManagement";
     import GenericResponseHandler from "@/use/helpers/GenericResponseHandler";
     import { Role } from "@/entities/Role";
-    import { useModelWrapper } from "@/use/helpers/ModelWrapper";
     import ErrorBag from "@/use/helpers/ErrorBag";
-    import Router from "@/use/router";
-    import SearchSelect from "@/components/common/SearchSelect.vue";
-    import SearchSelectOption from "@/use/helpers/SearchSelectOption";
 
     export default {
-        name: "LecturerSection",
+        name: "Lecturer",
         components: {
-            SearchSelect,
+            BaseSection,
+            SearchableSelect,
         },
         props: {
-            errorBag: {
-                required: true,
-                type: ErrorBag,
-            },
             lecturerId: {
                 type: String,
                 required: true,
             },
+            errorBag: {
+                type: Object as () => ErrorBag,
+                required: true,
+            },
         },
         emits: ["update:lecturerId"],
         setup(props: any, { emit }: any) {
             const lecturers = ref([] as Lecturer[]);
-            const currentLecturer = ref({} as Lecturer);
-            const lecturerFound = ref(false);
 
-            const optionsArray = ref([] as SearchSelectOption[]);
-            const selectedOption = ref({} as SearchSelectOption);
+            const currentSelection = ref({} as SearchSelectOption);
+
+            const selectableLecturers = computed(() =>
+                lecturers.value.map((l: Lecturer) => {
+                    return {
+                        value: l,
+                        display: createOptionString(l),
+                    } as SearchSelectOption;
+                })
+            );
 
             onBeforeMount(async () => {
                 await getLecturers();
```

Comment:
We may consider to prop a variable `const loading = ref(0)` from the parent component (course form) to the childs. Then the variable could be used as a loading components counter (by adding/substracting 1 on asynchronous call), such that the loading spinner within the parent component could be should by `v-if="loading > 0"`
Atm we have multiple child components that execute async calls like getLecturers or the store getter for the configuration without triggering any loading component

Reply:
Added a loading spinner to that section in case an API call takes that long :)